lib/ansible/modules/clustering/kubevirt/kubevirt_pvc.py

```diff
@@ -0,0 +1,204 @@
+      is simply C(strategic-merge).
+    choices:
+    - json
+    - merge
+    - strategic-merge
+    type: list
+  name:
+    description:
+      - Use to specify a PVC object name.
+    required: true
+    type: str
+  namespace:
+    description:
+      - Use to specify a PVC object namespace.
+    required: true
+    type: str
+  annotations:
+    description:
+      - Annotations attached to this object.
+      - U(https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/)
+    type: dict
+  labels:
+    description:
+      - Labels attached to this object.
+      - U(https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/)
+    type: dict
+  selector:
+    description:
+      - A label query over volumes to consider for binding.
+      - U(https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/)
+    type: dict
+  access_modes:
+    description:
+      - accessModes contains the desired access modes the volume should have.
+      - More info: U(https://kubernetes.io/docs/concepts/storage/persistent-volumes#access-modes)
+    type: list
+  resources:
+    description:
+      - Resources represents the minimum resources the volume should have.
+      - More info: U(https://kubernetes.io/docs/concepts/storage/persistent-volumes#resources)
+    type: dict
+  storage_class_name:
+    description:
+      - Name of the StorageClass required by the claim.
+      - More info: U(https://kubernetes.io/docs/concepts/storage/persistent-volumes#class-1)
+    type: str
+  volume_mode:
+    description:
+      - volumeMode defines what type of volume is required by the claim. Value of Filesystem is implied when not included in claim spec. This is an alpha feature of kubernetes and may change in the future.
+    type: str
+  volume_name:
+    description:
+      - volumeName is the binding reference to the PersistentVolume backing this claim.
+    type: str
+  data_transfer_type:
+    description:
+      - Upon creation, fill the volume with existing data.
+      - upload: upload data from local system. Needs to be followed up with usage of kubevirt_cdi_upload module.
+      - import: import data from remote system.
+      - clone: clone data from another PVC.
+      - TODO desc
+    choices:
+    - upload
+    - import
+    - clone
+    type: list
+  data_content_type:
+    description:
+      - Defines the content type of data being imported.
+      - I(kubevirt) denotes a virtual machine image. (Default)
+      - I(archive) denotes a tar archive.
+    choices:
+      - kubevirt
+      - archive
+  data_import_source:
+    description:
+      - Defaults to 'http' if missing of invalid.
+      - TODO… choices?
+  data_import_endpoint:
+    description:
+      - URL with the data. https or https is supported.
+  data_import_secret_name:
+    description:
+      - The name of the secret containing credentials for the end point
+  data_clone_source_namespace:
+    …
+  data_clone_source_pvc:
```

Comment:
Most of the annotations in CDI are storage.X, so storage.import.X, storage.upload.X. I didn't want to use `storage_` as a prefix, because there's already _spec.storageClassName_ in the object schema, which turns into `storage_class_name` in accordance with the typical ansible module param naming style. I figured `data_transfer_` prefix would've been the most correct, but that's a tad too long, so that got shortened to just `data_`.

The problem with just dropping `data_` is that there's something like `storage.contentType` annotation, which would then become `content_type` argument, which is a *very* bad argument name when it (currently) only affects some of the import functionality.

I do believe we should stick to a prefix. `data_` made sense for me as a shortened version of `data_transfer_` when `data_transfer_type` was present and required to initialize any kind of data transfer. But if we drop that argument, then `data_` becomes a vague prefix. It'd be great if somebody came up with something better. What comes to my mind:

1. `transfer_`
2. `dt_` (for data transfer)

Reply:
What about `cdi` prefix? Actually also what about having `k8_pvc` module and then create `kubevirt_pvc` with code reause and just adding `cdi` features?

Reply:
Looks like `cdi` prefix would give more information about why it is needed.